library/common-library/DEVELOPERS.md

```diff
@@ -0,0 +1,615 @@
+{{- $truncatedName := include "newrelic.common.naming.truncateToDNS" $nameToTruncate }}
+{{- $truncatedName }}
+{{- /* This should print: a-really-really-really-really-REALLY-long-string-that-should-be */ -}}
+```
+
+### `newrelic.common.naming.truncateToDNSWithSuffix`
+This template function is the same as the above but instead of receiving a string you should give a `dict`
+with a `name` and a `suffix`. This function will join them with a dash (`-`) and trim the `name` so the
+result of `name-suffix` is no more than 63 chars
+
+Usage:
+```mustache
+{{ $nameToTruncate := "a-really-really-really-really-REALLY-long-string-that-should-be-truncated-because-it-is-enought-long-to-brak-something"
+{{- $suffix := "A-NOT-SO-LONG-SUFFIX" }}
+{{- $truncatedName := include "truncateToDNSWithSuffix" (dict "name" $nameToTruncate "suffix" $suffix) }}
+{{- $truncatedName }}
+{{- /* This should print: a-really-really-really-really-REALLY-long-A-NOT-SO-LONG-SUFFIX */ -}}
+```
+
+
+
+## _labels.tpl
+### `newrelic.common.labels`, `newrelic.common.labels.selectorLabels` and `newrelic.common.labels.podLabels`
+These are functions that are used to label objects. They are configured by this `values.yaml`
+```yaml
+global:
+  podLabels: {}  # included in all the pods of all the charts that implement this library
+  labels: {}  # included in all the objects of all the charts that implement this library
+podLabels: {}  # included in all the pods of the chart that the chart writes is implementing
+labels: {}  # included in all the objects of the chart that the chart writes is implementing
+```
+
+label maps are merged from global to local values.
+
+And chart writer should use them like this:
+```mustache
+metadata:
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "newrelic.common.labels.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "newrelic.common.labels.podLabels" . | nindent 8 }}
+```
+
+`newrelic.common.labels.podLabels` includes `newrelic.common.labels.selectorLabels` automatically.
+
+
+
+## _priority-class-name.tpl
+### `newrelic.common.priorityClassName`
+Like almost everything in this library, it read global and local variables:
+```yaml
+global:
+  priorityClassName: ""
+priorityClassName: ""
+```
+
+Be careful: chart writers should put an empty string (or any kind of Helm falsiness) for this
+library to work properly. If in your values a non-falsy `priorityClassName` is found, the global
+one is going to be always ignored.
+
+Usage (example in a pod spec):
+```mustache
+spec:
+  {{- with include "newrelic.common.priorityClassName" . }}
+  priorityClassName: {{ . }}
+  {{- end }}
+```
+
+
+
+## _hostnetwork.tpl
+### `newrelic.common.hostNetwork`
+Like almost everything in this library, it read global and local variables:
```

Comment:
```suggestion
Like almost everything in this library, it reads global and local variables:
```

Reply:
also below there is the same typo

Reply:
Fixed in [#6216d9f](https://github.com/newrelic/helm-charts/pull/797/commits/6216d9f3f09fedf3717c1b7a457a887fd8230371)